### PR TITLE
Refactors Corgis and Other Doggos

### PR DESCRIPTION
**Pull request by Fox-McCloud**
From TG.

Refactors corgis a bit

changes:
- Equipping/unequipping items is handled by a `/datum/dog_fashion` that determines where it can be equipped and what, if any special behaviors a particular item has, for a corgi, when equipped; it's a lot better than snowflake checks
- Corgis are now repathed to /mob/living/simple_animal/dog
- Pugs are now a dog subtype; can now pet them
- Ian now has some persistence
  - If he's killed he'll start out the next shift as a puppy
  - If he breaks an age record, he'll gain a unique sprite until killed
- Added a rare chance for corgis to have void puppies (a tribute to animals being created in null space)
- Nar-Sie will convert Ian into a cult pup who will kill other pets

Not From TG
- Code is refactored to allow for all pets to have a specified collar type, though right now, it's still only corgis, cats, and a few others that can have a collar sprite
- Foxes are now a dog subtype, allowing them to properly receive pets and pats in addition to being generally doglike
- New generic dog mob
- All dogs can nibble and nom on food
- Borgis are no longer an Ian subtype (just a corgi subtype)
- Nar-Sie will absolutely not tolerate sacrificing doggos (as opposed to just corgis)

Fixes
- Fixes a runtime with eating/tasting null reagents as a corgi

:cl: Fox McCloud
add: Ian now has persistence; if he's killed, he'll start over life as a puppy--if he lives to a very long old age, he'll get a unique sprite
tweak: Pugs will now yipe on death and can bark
add: Adds new dog mob; a generic black dog
tweak: Foxes are now a dog subtype
add: Can manually yelp as a dog
tweak: All dog subtypes can now chase their tail and be given pats and pets. They all can additionally eat (and taste!) food
tweak: Nar-Sie will absolutely not tolerate sacrificing doggos
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
And Yes people, 65 doggone files to refactor corgis.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -0,0 +1,754 @@
...
+	var/dat = 	"<div align='center'><b>Inventory of [name]</b></div><p>"
+	dat += "<br><B>Head:</B> <A href='?src=[UID()];[inventory_head ? "remove_inv=head'>[inventory_head]" : "add_inv=head'>Nothing"]</A>"
+	dat += "<br><B>Back:</B> <A href='?src=[UID()];[inventory_back ? "remove_inv=back'>[inventory_back]" : "add_inv=back'>Nothing"]</A>"
+	dat += "<br><B>Collar:</B> <A href='?src=[UID()];[pcollar ? "remove_inv=collar'>[pcollar]" : "add_inv=collar'>Nothing"]</A>"
+
+	var/datum/browser/popup = new(user, "mob[UID()]", "[src]", 440, 250)
+	popup.set_content(dat)
+	popup.open()
+
+/mob/living/simple_animal/pet/dog/corgi/getarmor(def_zone, type)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Yes, corgis are actually impacted by armor they wear now.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+		if (shaved)
+			to_chat(user, "<span class='warning'>You can't shave this corgi, it's already been shaved!</span>")
+			return
+		if (nofur)
+			to_chat(user, "<span class='warning'>You can't shave this corgi, it doesn't have a fur coat!</span>")
+			return
+		user.visible_message("<span class='notice'>[user] starts to shave [src] using \the [O].", "<span class='notice'>You start to shave [src] using \the [O]...</span>")
+		if(do_after(user, 50, target = src))
+			user.visible_message("<span class='notice'>[user] shaves [src]'s hair using \the [O].</span>")
+			playsound(loc, 'sound/items/welder2.ogg', 20, TRUE)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: change to o.usesound

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 3]
```diff
@@ -0,0 +1,754 @@
...
+		return
+
+	//Removing from inventory
+	if(href_list["remove_inv"])
+		var/remove_from = href_list["remove_inv"]
+		switch(remove_from)
+			if("head")
+				if(inventory_head)
+					if(inventory_head.flags & NODROP)
+						to_chat(usr, "<span class='warning'>\The [inventory_head] is stuck too hard to [src] for you to remove!</span>")
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: needs a return

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 4]
```diff
@@ -0,0 +1,754 @@
...
+					inventory_head = null
+					update_corgi_fluff()
+					regenerate_icons()
+				else
+					to_chat(usr, "<span class='danger'>There is nothing to remove from its [remove_from].</span>")
+					return
+			if("back")
+				if(inventory_back)
+					if(inventory_back.flags & NODROP)
+						to_chat(usr, "<span class='warning'>\The [inventory_head] is stuck too hard to [src] for you to remove!</span>")
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: same

**Comment by KasparoVy** [comment]
Corgis be worth it

File: code/modules/admin/topic.dm [hunk 5]
```diff
@@ -1833,8 +1833,9 @@
 					P.can_collar = 1
 					P.faction = list("neutral")
 					var/obj/item/clothing/accessory/petcollar/C = new /obj/item/clothing/accessory/petcollar(P)
-					P.collar = C
+					P.pcollar = C
 					C.equipped(P)
+					P.regenerate_icons()
```
Comment (Fox-McCloud) [inline code/modules/admin/topic.dm]: should probably use the new equipping proc for collars

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 1 above)
Comment (KasparoVy) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: Death squad corgis

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 6]
```diff
@@ -572,4 +585,25 @@
...
 	if(AIStatus == AI_Z_OFF)
 		SSidlenpcpool.idle_mobs_by_zlevel[old_z] -= src
-		toggle_ai(initial(AIStatus))
\ No newline at end of file
+		toggle_ai(initial(AIStatus))
+
+/mob/living/simple_animal/proc/add_collar(obj/item/clothing/accessory/petcollar/P, mob/user)
+	if(QDELETED(P) || pcollar)
+		return
+	if(!user.unEquip(P))
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: should probably have a user check here for if we want to use this proc for non-users.

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 7]
```diff
@@ -453,32 +472,26 @@
...
 	W.plane = ABOVE_HUD_PLANE
 
 	switch(slot)
 		if(slot_collar)
-			collar = W
-			if(collar.tagname)
-				name = collar.tagname
-				real_name = collar.tagname
-			regenerate_icons()
+			add_collar(W)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: make sure this doesn't break/runtime.

**Comment by Fox-McCloud** [comment]
Thoughts: Foxes could probably be a dog subtype, though their sounds would be odd, until good quality ones were added.

It really wouldn't do much other than allow them to eat food and and get hearts when petted.

**Comment by taukausanake** [comment]
Is IconDiff bot broken? Going to need to post images for those of us who don't want to code dive

**Comment by KasparoVy** [comment]
'Narsian' is spooky and 'Original' looks sick. Also @Fox-McCloud wouldn't you be able to set their sounds to 'null' so they're as quiet as usual?

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 8]
```diff
@@ -1,669 +0,0 @@
...
-				name = "Lieutenant [real_name]"
-				desc = "Can actually be trusted to not run off on his own."
-				valid = 1
-
-			if(/obj/item/clothing/head/helmet/space/deathsquad)
-				name = "Trooper [real_name]"
-				desc = "That's not red paint. That's real corgi blood."
-				valid = 1
-
-			if(/obj/item/clothing/suit/space/hardsuit/engine)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm]: Try to re-add this one.

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 9]
```diff
@@ -1,669 +0,0 @@
...
-				name = "Space Explorer [real_name]"
-				desc = "That's one small step for a corgi. One giant yap for corgikind."
-				valid = 1
-				mutations.Add(BREATHLESS)
-				atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
-				minbodytemp = 0
-
-			if(/obj/item/clothing/mask/fakemoustache)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm]: Try to re-add this one.

**Comment by JKnutson101** [comment]
These changes are very fun and I look forward to them! Maybe this is outside of the scope of this PR, but have you considered turning corgi's that get nanomachines turn into borgi's? I've had a few corgis eat Roburgers before and it would be really funny if they were turned into borgis because of it.

**Comment by Fox-McCloud** [comment]
Simple animals don't process reagents and never will; at that point, they're carbons, which is a whole other mess of things.

Adding a snowflake check just for when corgis eat things (and if a reagent container they eat happens to have a particular reagent) is the very definition of snowflake.

Not really worth it.

File: code/modules/admin/topic.dm [hunk 10]
```diff
@@ -358,8 +358,7 @@
 			if("robot")				M.change_mob_type( /mob/living/silicon/robot , null, null, delmob, 1 )
 			if("cat")				M.change_mob_type( /mob/living/simple_animal/pet/cat , null, null, delmob, 1 )
 			if("runtime")			M.change_mob_type( /mob/living/simple_animal/pet/cat/Runtime , null, null, delmob, 1 )
-			if("corgi")				M.change_mob_type( /mob/living/simple_animal/pet/corgi , null, null, delmob, 1 )
-			if("ian")				M.change_mob_type( /mob/living/simple_animal/pet/corgi/Ian , null, null, delmob, 1 )
```
Comment (Fox-McCloud) [inline code/modules/admin/topic.dm]: We really shouldn't be making additional Ians now; if you're going to spawn additional corgis, it should be the base type, *NOT* Ian.

File: code/modules/mob/transform_procs.dm [hunk 11]
```diff
@@ -184,8 +184,7 @@
 	for(var/t in bodyparts)	//this really should not be necessary
 		qdel(t)
 
-	var/mob/living/simple_animal/pet/corgi/new_corgi = new /mob/living/simple_animal/pet/corgi (loc)
-	new_corgi.a_intent = INTENT_HARM
```
Comment (Fox-McCloud) [inline code/modules/mob/transform_procs.dm]: Why spawn corgis on harm intent? RUDE

**Comment by variableundefined** [comment]
It appears the dog days of summer is not over yet

**Pull request merged**